### PR TITLE
Use fuzztest targets for oss-fuzz.

### PR DESCRIPTION
**Pull request by vrabaud**
To test locally, follow the instructions at the beginning of the file.

Also: Limit CIFuzz to when a PR is merged
This can be revisited at some point but first:
- let's make sure new oss-fuzz targets are built and created
on the site
- let's make sure we fix all fuzzer bugs if any
- let's see how long the job really takes

### CODE REVIEW

File: tests/oss-fuzz/build.sh [hunk 1]
```diff
@@ -19,19 +19,76 @@
...
 
 # build libavif
 mkdir build
 cd build
-cmake -G Ninja -DBUILD_SHARED_LIBS=0 -DAVIF_CODEC_DAV1D=1 -DAVIF_LOCAL_DAV1D=1 ..
+cmake -G Ninja -DBUILD_SHARED_LIBS=OFF -DAVIF_CODEC_AOM=ON -DAVIF_CODEC_DAV1D=ON \
+      -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DAVIF_ENABLE_WERROR=OFF \
+      -DAVIF_LOCAL_AOM=ON -DAVIF_LOCAL_DAV1D=ON -DAVIF_LOCAL_FUZZTEST=ON \
+      -DAVIF_LOCAL_GTEST=ON -DAVIF_LOCAL_JPEG=ON -DAVIF_LOCAL_ZLIBPNG=ON \
+      -DAVIF_BUILD_TESTS=ON -DAVIF_ENABLE_GTEST=ON -DAVIF_ENABLE_FUZZTEST=ON ..
```
Comment (y-guyon) [inline tests/oss-fuzz/build.sh:57]: Note: No libyuv, no libsharpyuv for now.

File: tests/oss-fuzz/build.sh [hunk 2]
```diff
@@ -19,19 +19,76 @@
...
+
+# build fuzztests
+# The following is taken from https://github.com/google/oss-fuzz/blob/31ac7244748ea7390015455fb034b1f4eda039d9/infra/base-images/base-builder/compile_fuzztests.sh#L59
+# Iterate the fuzz binaries and list each fuzz entrypoint in the binary. For
+# each entrypoint create a wrapper script that calls into the binaries the
+# given entrypoint as argument.
+# The scripts will be named:
+# {binary_name}@{fuzztest_entrypoint}
+FUZZ_TEST_BINARIES_OUT_PATHS=`ls ./tests/avif_fuzztest_*`
+#cp ./bin/opencv_fuzz_* $OUT/
```
Comment (y-guyon) [inline tests/oss-fuzz/build.sh:74]: Leftover?

**Review by y-guyon — APPROVED** [review]

**Review by y-guyon — APPROVED** [review]

File: tests/oss-fuzz/build.sh (hunk 1 above)
Comment (vrabaud) [reply tests/oss-fuzz/build.sh]: Thx, fixed and added

File: tests/oss-fuzz/build.sh (hunk 2 above)
Comment (vrabaud) [reply tests/oss-fuzz/build.sh]: removed

**Comment by vrabaud** [comment]
Not merging yet because that increases the CI for fuzz because it launches this script, which builds a looot of dependencies. Looking into caching that,

**Comment by vrabaud** [comment]
The build now "only" takes 10 minutes. Still, I am removing it from CIFuzz for now because it hangs due to the targets not being officially create on oss-fuzz. Let's wait for them to be there, and we can re-enable the CI.

**Comment by vrabaud** [comment]
Link to official doc on how to only allow CI for merges: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-when-a-pull-request-merges

let's keep the concurrency for now in case we re-enable the CI per open pull request.

**Review by y-guyon — APPROVED** [review]

**Pull request merged**